File: frontend/src/api/learning.ts
import api from "./api";

export interface LearningCandidatePayload {
  source_submission_id?: string;
  complaint_id?: number;
  complaint_title: string;
  complaint_description: string;
  original_category: string;
  original_department: string;
  original_manager: string;
  original_recommendation: string;
  verification_rating: "Yes" | "Partially" | "No";
  verification_score: number;
  employee_comments: string;
  corrected_category?: string;
  corrected_department?: string;
  corrected_manager?: string;
  corrected_action?: string;
}

export interface LearningCandidateResponse extends LearningCandidatePayload {
  id: number;
  reviewer_notes?: string;
  status: "Needs Review" | "Approved" | "Rejected";
  created_at: string;
  reviewed_at?: string;
}

export interface LearningDecisionPayload {
  action: "approve" | "reject";
  corrected_category?: string;
  corrected_department?: string;
  corrected_manager?: string;
  corrected_action?: string;
  reviewer_notes?: string;
}

export const createLearningCandidate = async (
  data: LearningCandidatePayload
): Promise<LearningCandidateResponse> => {
  const response = await api.post("/learning/candidates", data);
  return response.data;
};

export const getLearningCandidates = async (
  status?: LearningCandidateResponse["status"]
): Promise<LearningCandidateResponse[]> => {
  const response = await api.get("/learning/candidates", {
    params: status ? { status } : undefined
  });
  return response.data;
};

export const decideLearningCandidate = async (
  id: number,
  data: LearningDecisionPayload
): Promise<LearningCandidateResponse> => {
  const response = await api.patch(`/learning/candidates/${id}/decision`, data);
  return response.data;
};

File: backend-ai/app/schemas/learning.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class LearningCandidateCreate(BaseModel):
    source_submission_id: Optional[str] = None
    complaint_id: Optional[int] = None
    complaint_title: str
    complaint_description: str
    original_category: str
    original_department: str
    original_manager: str
    original_recommendation: str
    verification_rating: str
    verification_score: float
    employee_comments: str
    corrected_category: Optional[str] = None
    corrected_department: Optional[str] = None
    corrected_manager: Optional[str] = None
    corrected_action: Optional[str] = None


class LearningCandidateDecision(BaseModel):
    action: str
    corrected_category: Optional[str] = None
    corrected_department: Optional[str] = None
    corrected_manager: Optional[str] = None
    corrected_action: Optional[str] = None
    reviewer_notes: Optional[str] = None


class LearningCandidateResponse(BaseModel):
    id: int
    source_submission_id: Optional[str] = None
    complaint_id: Optional[int] = None
    complaint_title: str
    complaint_description: str
    original_category: str
    original_department: str
    original_manager: str
    original_recommendation: str
    verification_rating: str
    verification_score: float
    employee_comments: str
    corrected_category: Optional[str] = None
    corrected_department: Optional[str] = None
    corrected_manager: Optional[str] = None
    corrected_action: Optional[str] = None
    reviewer_notes: Optional[str] = None
    status: str
    created_at: datetime
    reviewed_at: Optional[datetime] = None

    model_config = ConfigDict(
        from_attributes=True
    )
